### PR TITLE
New entites XML representation

### PR DESCRIPTION
**Pull request by julianbermudez**
XML representation for data_writer and data_reader has been added in the FastRTPS' XML Paser, so examples have been changed according with the new feature.
On the other hand, a new tree test for reference representation was added.

### CODE REVIEW

File: DEFAULT_FASTRTPS_PROFILES.xml [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
-        <kind>KEEP_LAST</kind>
-        <depth>5</depth>
-      </historyQos>
-      <durability>
-        <kind>TRANSIENT_LOCAL</kind>
-      </durability>
+    <participant profile_name="default_xrce_participant">
+        <rtps>
+            <builtin>
+                <leaseDuration>
```
Comment (richiware) [inline DEFAULT_FASTRTPS_PROFILES.xml]: Simplify xml. Remove leaseDuration.

File: DEFAULT_FASTRTPS_PROFILES.xml [hunk 2]
```diff
@@ -1,53 +1,53 @@
...
+            </builtin>
+            <name>default_xrce_participant</name>
+        </rtps>
+    </participant>
+    <data_writer profile_name="shapetype_data_writer">
+        <topic>
+            <kind>WITH_KEY</kind>
+            <name>Square</name>
+            <dataType>ShapeType</dataType>
+            <historyQos>
```
Comment (richiware) [inline DEFAULT_FASTRTPS_PROFILES.xml]: Simplify xml. Remove history and durability. uses the default.

File: DEFAULT_FASTRTPS_PROFILES.xml [hunk 3]
```diff
@@ -1,53 +1,53 @@
...
+                <kind>TRANSIENT_LOCAL</kind>
+            </durability>
+        </topic>
+    </data_writer>
+    <data_reader profile_name="shapetype_data_reader">
+        <topic>
+            <kind>WITH_KEY</kind>
+            <name>Square</name>
+            <dataType>ShapeType</dataType>
+            <historyQos>
```
Comment (richiware) [inline DEFAULT_FASTRTPS_PROFILES.xml]: Simplify xml. Remove history and durability. uses the default.

File: DEFAULT_FASTRTPS_PROFILES.xml [hunk 4]
```diff
@@ -1,53 +1,53 @@
...
+            <durability>
+                <kind>TRANSIENT_LOCAL</kind>
+            </durability>
+        </topic>
+    </data_reader>
+    <topic profile_name="shapetype_topic">
+        <kind>WITH_KEY</kind>
+        <name>Square</name>
+        <dataType>ShapeType</dataType>
+        <historyQos>
```
Comment (richiware) [inline DEFAULT_FASTRTPS_PROFILES.xml]: Simplify xml. Remove history and durability. uses the default.

File: test/DEFAULT_FASTRTPS_PROFILES.xml [hunk 5]
```diff
@@ -1,41 +1,46 @@
...
-        <kind>KEEP_LAST</kind>
-        <depth>5</depth>
-      </historyQos>
-      <durability>
-        <kind>TRANSIENT_LOCAL</kind>
-      </durability>
+    <participant profile_name="default_xrce_participant">
+        <rtps>
+            <builtin>
+                <leaseDuration>
```
Comment (richiware) [inline test/DEFAULT_FASTRTPS_PROFILES.xml]: Simplify xml. Remove leaseDuration. uses the default.

File: test/DEFAULT_FASTRTPS_PROFILES.xml [hunk 6]
```diff
@@ -1,41 +1,46 @@
...
+            <name>default_xrce_participant</name>
+        </rtps>
+    </participant>
+    <data_writer profile_name="shapetype_data_writer">
+        <topic>
+            <kind>WITH_KEY</kind>
+            <name>Square</name>
+            <dataType>ShapeType</dataType>
+            <historyQos>
+                <kind>KEEP_LAST</kind>
```
Comment (richiware) [inline test/DEFAULT_FASTRTPS_PROFILES.xml]: Simplify xml. Remove history and durability. uses the default.

File: test/DEFAULT_FASTRTPS_PROFILES.xml [hunk 7]
```diff
@@ -1,41 +1,46 @@
...
+                <kind>TRANSIENT_LOCAL</kind>
+            </durability>
+        </topic>
+    </data_writer>
+    <data_reader profile_name="shapetype_data_reader">
+        <topic>
+            <kind>WITH_KEY</kind>
+            <name>Square</name>
+            <dataType>ShapeType</dataType>
+            <historyQos>
```
Comment (richiware) [inline test/DEFAULT_FASTRTPS_PROFILES.xml]: Simplify xml. Remove history and durability. uses the default.

File: test/blackbox/tree/TreeTests.cpp [hunk 8]
```diff
@@ -56,10 +56,22 @@ TEST_F(TreeTests, XMLTree)
      * Create participant.
      */
     dds::xrce::ObjectVariant object_variant;
+    std::string participant_xml = "<dds>"
+                                      "<participant>"
+                                          "<rtps>"
+                                              "<builtin>"
+                                                  "<leaseDuration>"
```
Comment (richiware) [inline test/blackbox/tree/TreeTests.cpp]: Simplify xml. Remove leaseDuration. uses the default.

File: test/blackbox/tree/TreeTests.cpp [hunk 9]
```diff
@@ -95,10 +112,25 @@ TEST_F(TreeTests, XMLTree)
...
      * Create datawriter.
      */
-    std::string datawriter_xml = "<profiles><publisher profile_name=\"default_xrce_publisher_profile\"><topic><kind>NO_KEY</kind><name>HelloWorldTopic</name><dataType>HelloWorld</dataType><historyQos><kind>KEEP_LAST</kind><depth>5</depth></historyQos><durability><kind>TRANSIENT_LOCAL</kind></durability></topic></publisher></profiles>";
+    std::string datawriter_xml = "<dds>"
+                                     "<data_writer>"
+                                         "<topic>"
+                                             "<kind>NO_KEY</kind>"
+                                             "<name>HelloWorldTopic</name>"
+                                             "<dataType>HelloWorld</dataType>"
+                                             "<historyQos>"
```
Comment (richiware) [inline test/blackbox/tree/TreeTests.cpp]: Simplify xml. Remove history and durability. uses the default.

File: test/blackbox/tree/TreeTests.cpp [hunk 10]
```diff
@@ -121,10 +153,129 @@ TEST_F(TreeTests, XMLTree)
...
      * Create datareader.
      */
-    std::string datareader_xml = "<profiles><subscriber profile_name=\"default_xrce_subscriber_profile\"><topic><kind>NO_KEY</kind><name>HelloWorldTopic</name><dataType>HelloWorld</dataType><historyQos><kind>KEEP_LAST</kind><depth>5</depth></historyQos><durability><kind>TRANSIENT_LOCAL</kind></durability></topic></subscriber></profiles>";
+    std::string datareader_xml = "<profiles>"
+                                     "<data_reader>"
+                                         "<topic>"
+                                             "<kind>NO_KEY</kind>"
+                                             "<name>HelloWorldTopic</name>"
+                                             "<dataType>HelloWorld</dataType>"
+                                             "<historyQos>"
```
Comment (richiware) [inline test/blackbox/tree/TreeTests.cpp]: Simplify xml. Remove history and durability. uses the default.

File: DEFAULT_FASTRTPS_PROFILES.xml (hunk 1 above)
Comment (julianbermudez) [reply DEFAULT_FASTRTPS_PROFILES.xml]: Done!

File: DEFAULT_FASTRTPS_PROFILES.xml (hunk 2 above)
Comment (julianbermudez) [reply DEFAULT_FASTRTPS_PROFILES.xml]: Done!

File: DEFAULT_FASTRTPS_PROFILES.xml (hunk 3 above)
Comment (julianbermudez) [reply DEFAULT_FASTRTPS_PROFILES.xml]: Done!

File: DEFAULT_FASTRTPS_PROFILES.xml (hunk 4 above)
Comment (julianbermudez) [reply DEFAULT_FASTRTPS_PROFILES.xml]: Done!

**Pull request merged**